test-workflows/tasks/generators/clusterloader/load.yaml

```diff
@@ -0,0 +1,127 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: clusterloader2
+spec:
+  description: >-
+    clusterloader2 task to run various types of cl2 tests on a given cluster.
+  params:
+  - name: giturl
+    description: "git url to clone the package"
+    default: https://github.com/kubernetes/perf-tests.git
+  - name: version
+    description: golang version to use for builds
+    default: "latest"
+  - name: GOOS
+    description: "running program's operating system target"
+    default: linux
+  - name: GOARCH
+    description: "running program's architecture target"
+    default: amd64
+  - name: GO111MODULE
+    description: "value of module support"
+    default: auto
+  - name: GOCACHE
+    description: "Go caching directory path"
+    default: ""
+  - name: GOMODCACHE
+    description: "Go mod caching directory path"
+    default: ""
+  - name: GOPROXY
+    default: direct
+  - name: cluster-name
+    description: name of the cluster 
+    default: hakuna-matata
+  - name: region
+    description: region of the cluster
+    default: us-west-2
+  - name: endpoint
+    description: "aws eks enpoint"
+    default: https://api.us-west-2.wesley.amazonaws.com 
+  - name: nodes-per-namespace
+    description: "nodes per namespace to get created for load test "
+    default: 10
+  - name: cl2-load-test-throughput
+    description: " throughput used for mutate operations"
+    default: 10
+  - name: pods-per-node
+    description: "pod density"
+    default: 10
+  - name: nodes
+    description: "number of dataplane nodes to run the load test against"
+    default: 100
+  workspaces:
+  - name: source
+  - name: config
+  - name: secrets
+    mountPath: /tekton/home/.aws
+  steps:
+  - name: write-kubeconfig
+    image: amazon/aws-cli
+    workingDir: $(workspaces.config.path)
+    script: |
+      aws sts get-caller-identity
+      # get the kube config if not present in default location using 
+      if [ ! -e $(workspaces.config.path)/kubeconfig ];then
+          echo "kube config"
+          aws eks --endpoint $(params.endpoint) update-kubeconfig --name $(params.cluster-name) --region $(params.region)
```

Comment:
Shouldn't we be passing in a kubeconfig from our create-cluster step?

Reply:
We are indeed passing, that's what the `if` condition is. It gives priority to the kubeconfig file passed from `create` cluster. But also gives an option of running tests on an adhoc cluster that's already created and run test against it too.

Reply:
Let's simplify this entirely and require that this task is chained with a cluster setup command. I'm happy to have a setup task that is just an `aws eks update-kubeconfig` command.